addons/resource/models/resource_calendar_leaves.py

```diff
@@ -53,10 +52,14 @@ def _compute_company_id(self):
 
     @api.depends('date_from')
     def _compute_date_to(self):
-        user_tz = timezone(self.env.user.tz or self._context.get('tz') or self.company_id.resource_calendar_id.tz or 'UTC')
-        for leave in self:
-            date_to_tz = user_tz.localize(leave.date_from) + relativedelta(hour=23, minute=59, second=59)
-            leave.date_to = date_to_tz.astimezone(utc).replace(tzinfo=None)
+        """This compute serves only UX purpose within hr_holidays module (in specific view afaik).
+        When user inputs a public holiday starting day, he doesn't need to fill end date (as it is set to the end of the same day).
+        Dear Refactorer: Please remove this function from the resource module if you can (should be in hr_holidays)"""
+        if self._context.get('params', {}).get('menu_id') == self.env.ref('hr_holidays.hr_holidays_public_time_off_menu_configuration').id:
```

Comment:
Instead of fetching & comparing `menu_id` using context, I think a better approach would be to add something like `public_holiday: True` to the context here, and check for that: https://github.com/odoo/odoo/blob/254d6a71e8d5e7bdf8b302d774e024646921b72d/addons/hr_holidays/views/resource_views.xml#L80-L82

Reply:
Thanks for having a look, that's a great idea.

Although that would probably require additional test